### PR TITLE
New Queries related with Trips

### PR DESCRIPTION
**Pull request by bujosa**
### Changelog;
- Create self publish trip  for managers
- Create publish trip for admin
- Create cancel trip for admin
- Create self cancel trip for managers
- Add new validators, mongo id validator, description and list entitities
- Create list trips query
- Create self trips
- Create list trips 
- Add trip on application
- Add test application inputs
- Add generate ticket, and price functions.

### CODE REVIEW

File: apps/trips/src/common/trip/graphql/inputs/trips/create-trip.input.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ export class CreateTripInput extends ValidationInput {
 
   manager?: string;
 
+  ticket?: string;
```
Comment (goritm) [inline apps/trips/src/common/trip/graphql/inputs/trips/create-trip.input.ts:35]:
```suggestion
  ticker?: string;
```

File: apps/trips/src/common/trip/trip.service.ts [hunk 2]
```diff
@@ -2,10 +2,119 @@ import { Service } from '@shared/data/classes/service.class';
...
   }
+
+  public async createEntity(
+    createEntityInput: CreateTripInput,
+    jwtPayload?: JwtPayload
+  ): Promise<Trip> {
+    const { manager, stages } = createEntityInput;
+
+    createEntityInput.manager = manager ? manager : jwtPayload.id;
+    createEntityInput.ticket = generateTicket();
```
Comment (goritm) [inline apps/trips/src/common/trip/trip.service.ts:33]: Is is ticket or ticker?

**Review by goritm — APPROVED** [review]
Looks great! Left some questions about the trip's ticket, but other than that I'll aprove it

File: apps/trips/src/common/trip/graphql/inputs/trips/create-trip.input.ts (hunk 1 above)
Comment (bujosa) [reply apps/trips/src/common/trip/graphql/inputs/trips/create-trip.input.ts:35]: ticket

**Pull request merged**